### ISSUE TITLE
PascalVOC XML Contain Multiple Distinct Objects, But Want to Train One Object

### ISSUE THREAD

**Issue by mmortazavi** [EdjeElectronics/TensorFlow-Object-Detection-API-Tutorial-Train-Multiple-Objects-Windows-10#342]
I have used LabelImg to create bounding boxes for my objects. Each image has two distinct objects (i.e. classes). I have tried training on both objects. and it seems it works. Now I would like to train ONLY based on one of the objects (single class). Obviously I want to keep my original PascalVOC XML as it is, and just to ignore the other class during training. 

I have tried to eliminate having the other class  in 'generate_tfrecord.py' for generating the train.record and test.record files, but I could not! Apparently all classes that are part of the 'train_labels.csv' and 'test_labels.csv' needs to be mentioned explicitly. 

The only places I then thought I can exclude the other class was in 'labelmap.pbtxt' and in having ONLY 'num_classes: 1' in 'faster_rcnn_inception_v2_pets.config'! However the training results in weird losses. After 500 Epochs the loss drops from 1.2 to small values around 0.1, then there is a abrupt increase like 543467 in loss, and present in all losses. Again it drops and goes on dropping and then again I will see an increase (quite large and non sense). And this weird behavior doesn't change even after 150K epochs. 

The only thing I suspected was the only thing I changed, the number of classes. While all classes being present in PascalVOC XML, record files as well as 'train_labels.csv' and 'test_labels.csv' still contained both classes. I believe this has caused confusion during training and resulted in such weird behavior. 

What can I do in this case? Should I go back to LabelImg  and remove the other object bounding boxes and generate the PascalVOC XML again? I hope not, as this seems really not necessary.

**Comment by AeroWRX** [comment]
Have you just tried to restart from STEP 2c. and delete all the necessary files and recreate it from scratch?

**Comment by mmortazavi** [comment]
Well, yes and no. I deleted all the necessary files STEP 2c, and did STEP 2e but skipped through the rest including STEP 2d (obviously I have had the installs) and STEP 2f! I wonder now whether STEP 2f needs to be repeated for Protobuf files, which are used by TensorFlow to configure model and training parameters?

I am starting to think maybe it does. Anyone can confirm whether STEP 2f (compile Protobuf files) needs to be repeated whenever changing configs, labels etc.? I need to test it though. 

The confusion comes from the fact that the tutorial starts with the example that is given in the this git repo (for Cat and Dog), and it continues explaining how to do it in your own dataset (herein for Cards). But switching from Cat/Dogs to Cards, the tutorial doesn't repeat the STEP 2f, thus I thought it is not needed!
